src/command/root-command.ts

```diff
@@ -0,0 +1,40 @@
+import Bee from '@ethersphere/bee-js'
+import { ExternalOption } from 'furious-commander'
+
+/**
+ * This class can be parent of the Commands to handle root options of the CLI
+ */
+export class RootCommand {
+  // CLI FIELDS
+
+  @ExternalOption('https')
+  private https!: boolean
+
+  @ExternalOption('bee-host')
+  private beeHost!: string
+
+  @ExternalOption('bee-api-port')
+  private beeApiPort!: number
+
+  // CLASS FIELDS
+
+  /** assembled API URL of Bee */
+  public beeApiUrl!: string
+
+  public bee!: Bee
+
+  /** Gives back URL string */
+  protected assembleEndpoint(https: boolean, beeHost: string, port: number): string {
+    return `${https ? 'https' : 'http'}://${beeHost}:${port}`
+  }
+
+  /**
+   * Init Root command fields
+   *
+   * if BEE_API_URL environment variable has been set the CLI will use that connection string
+   */
+  protected init(): void {
+    this.beeApiUrl = process.env.BEE_API_URL || this.assembleEndpoint(this.https, this.beeHost, this.beeApiPort)
```

Comment:
I think we should make something similar in `bee-js` as well later, because there could be operations on the debug API port, where `bee-host` is the same for both urls. Therefore, I think it's easier to define in this way and also it emphasises the encrypted connection with `https` option.

I rather not provide for every option an environment variable possibility to set, not only because it's a lot, also later could happen overlappings.

Reply:
I'd have to agree with @agazso . If we want to enforce and stress `https` then we can throw warning message. I'd be all for just providing `beeApiUrl` directly